code/game/machinery/exonet_node.dm

```diff
@@ -0,0 +1,146 @@
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "exonet_node"
+	idle_power_usage = 25
+	var/on = 1
+	var/toggle = 1
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/exonet_node
+	max_integrity = 300
+	integrity_failure = 100
+	armor = list(melee = 20, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 50, acid = 70)
+	var/opened = 0
+
+	var/list/logs = list() // Gets written to by exonet's send_message() function.
+
+// Proc: New()
+// Parameters: None
+// Description: Adds components to the machine for deconstruction.
+/obj/machinery/exonet_node/map/New()
+	..()
+
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/machine/exonet_node(src)
+	component_parts += new /obj/item/stock_parts/subspace/ansible(src)
+	component_parts += new /obj/item/stock_parts/subspace/filter(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/stock_parts/subspace/crystal(src)
+	component_parts += new /obj/item/stock_parts/subspace/treatment(src)
+	component_parts += new /obj/item/stock_parts/subspace/treatment(src)
+	component_parts += new /obj/item/stack/cable_coil(src, 2)
+	RefreshParts()
+
+	desc = "This machine is exonet node."
+
+// Proc: update_icon()
+// Parameters: None
+// Description: Self explanatory.
+/obj/machinery/exonet_node/update_icon()
+	if(on)
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]_off"
+
+// Proc: update_power()
+// Parameters: None
+// Description: Sets the device on/off and adjusts power draw based on stat and toggle variables.
+/obj/machinery/exonet_node/proc/update_power()
+	if(toggle)
+		if(stat & (BROKEN|NOPOWER|EMPED))
+			on = 0
+			idle_power_usage = 0
+		else
+			on = 1
+			idle_power_usage = 2500
+	else
+		on = 0
+		idle_power_usage = 0
+	update_icon()
+
+// Proc: emp_act()
+// Parameters: 1 (severity - how strong the EMP is, with lower numbers being stronger)
+// Description: Shuts off the machine for awhile if an EMP hits it.  Ion anomalies also call this to turn it off.
+/obj/machinery/exonet_node/emp_act(severity)
+	if(!(stat & EMPED))
+		stat |= EMPED
+		var/duration = (300 * 10)/severity
+		spawn(rand(duration - 20, duration + 20))
+			stat &= ~EMPED
+	update_icon()
+	..()
+
+// Proc: process()
+// Parameters: None
+// Description: Calls the procs below every tick.
+/obj/machinery/exonet_node/process()
+	update_power()
+
+// Proc: attackby()
+// Parameters: 2 (I - the item being whacked against the machine, user - the person doing the whacking)
+// Description: Handles deconstruction.
+/obj/machinery/exonet_node/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/screwdriver))
+		default_deconstruction_screwdriver(user, I)
+	else if(istype(I, /obj/item/crowbar))
+		default_deconstruction_crowbar(user, I)
+	else
+		..()
```

Comment:
too many tabs in?

Reply:
Feels like it's just Github being retarded with tab size once again.